Added Epson Workforce printer sensor

ThaStealth opened the pull request:
## Description:

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#9200

## Example entry for `configuration.yaml` (if applicable):
```yaml
   - platform: epsonworkforce
     host: 192.168.0.123
     monitored_conditions:
     - black     
     - yellow
     - magenta
     - cyan
     - clean      
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] [_The manifest file_][manifest-docs] has all fields filled out correctly ([example][ex-manifest]).
  - [X] New dependencies have been added to `requirements` in the manifest ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [X] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-manifest]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json
[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json#L5
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23
[manifest-docs]: https://developers.home-assistant.io/docs/en/development_checklist.html#_the-manifest-file_

ThaStealth committed:
Added Epson Workforce printer sensor

ThaStealth committed:
Added Epson Workforce printer sensor

exxamalte committed:
Instituto Geográfico Nacional Sismología (Earthquakes) Feed platform (#22696)

* wip

* added new requirements

* fixed tests

* improved test coverage

* added feed entry's title

* added manifest

* updated codeowners

* generated requirements

ThaStealth committed:
Update sensor.py

ThaStealth committed:
Update sensor.py

fabaff commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+    host = config.get(CONF_HOST)
+
+    from epsonprinter_pkg.epsonprinterapi import EpsonPrinterAPI
+    api = EpsonPrinterAPI(host)
```
If the printer is offline then we should raise `PlatformNotReady`.

fabaff commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+
+    def __init__(self, hass, api, variable):
+        """Initialize a cartridge sensor."""
+        self._hass = hass
```
Not needed.

ThaStealth replied:
Okay, copied it from another component

fabaff commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+
+
+class EpsonPrinterCartridge(Entity):
+    """Representation of a cartdige sensor."""
```
Typo

ThaStealth replied:
whoops, changed it

fabaff commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+    'clean': ['Inklevel Cleaning', '%', 'mdi:water'],
+}
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST, default=DEFAULT_IP): cv.string,
```
If it's required then there should be no default.

ThaStealth replied:
Okay, removed it

fabaff commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+        self._var_name = variable_info[0]
+        self._var_id = variable
+        self._var_unit = variable_info[1]
+        self._var_icon = variable_info[2]
```
There is no need to prefix the name with `_var`.

```suggestion
        self._icon = variable_info[2]
```

ThaStealth replied:
changed it

ThaStealth committed:
Changes after review

ThaStealth committed:
Changes after review

ThaStealth committed:
Fixed lint error

ThaStealth committed:
Fixed lint error

ThaStealth committed:
Update sensor.py

ThaStealth committed:
Update sensor.py

adrum committed:
Update HAP-python to 2.5.0 (#23031)

bachya committed:
Remove expired 17track.net packages from entity registry (#23001)

* Remove expired 17track.net packages from entity registry

* Reverse order

cgarwood committed:
Replace get_platform (#23014)

* Update Z-Wave to use async_get_integration

* Change load method per PR comments

* update tests

balloob committed:
Convert service helper to use async_get_integration (#23023)

* Convert service helper to use async_get_integration

* Fix tests

cgtobi committed:
Remove dependencies and requirements (#23024)

* Remove dependencies and requirements

* Revert "Remove dependencies and requirements"

This reverts commit fe7171b4cd30889bad5adc9a4fd60059d05ba5a7.

* Remove dependencies and requirements

* Revert "Remove dependencies and requirements"

This reverts commit 391355ee2cc53cbe6954f940062b18ae34b05621.

* Remove dependencies and requirements

* Fix flake8 complaints

* Fix more flake8 complaints

* Revert non-component removals

balloob committed:
Migrate check-config to use get_integration (#23026)

* Migrate check-config to use get_integration

* ImportError

elupus committed:
Move Broadlink services to component (#21465)

* Register services in broadlink domain

* Add tests for broadlink services

* Resolve review comments

* One more review fix

* Restore auth retry

* Drop unused constants

* Fix flake8 errors

dgomes committed:
Bandage telegram bot (#23022)

* Bandage

* lint

* move everything into __init__.py

* fix lint

bachya committed:
Update RainMachine sensors in parallel (#23057)

balloob committed:
Only load stream when av package available (#23058)

balloob committed:
Convert translation helper to use async_get_integration (#23054)

* Convert translation helper to use async_get_integration

* Simplify after comments

* Lint

* Fix typing

* Typo

awarecan committed:
Allow aws credential skip validation (#22991)

* Allow aws credential skip validation

* Don't validate the auto-created default profile

BKPepe committed:
Upgrade python-slugify to 3.0.2 (#22997)

c-soft committed:
Add Satel_integra switchable outputs and multiple partitions (#21992)

* Added editable outputs and multiple zones.

* Updated requirements_all.txt

* Linter fixes.

* Post-review changes

* Fixed too many lines separation error

* Passing satel controller as parameter to entities.

* Fixed linter error.

* Fixed forgotten requirements update.

* Fixed satel_integra version (again!?!)

* Fixed manifest.json.

* Fixed passing non-serializable controller

* Removed unnecessary isinstance check.

* Post review changes

Adminiuga committed:
Bugfix ZHA device_removed() handler. (#23074)

Swamp-Ig committed:
Fix websocket connection sensor (#22923)

* Fix for #22890

* Singleton count

elupus committed:
Validate data packet format on config validation (#23062)

bachya committed:
Create decorator to check service permissions (#22667)

* Create decorator to check service permissions

* Typing

* Linting

* Member comments

* Linting

* Member comments

* Updated import

* Owner comments

* Linting

* Linting

* More work

* Fixed tests

* Removed service helper tests in RainMachine

* Linting

* Owner comments

* Linting

* Owner comments

Co-Authored-By: bachya <bachya1208@gmail.com>

balloob committed:
Validate component usage (#23037)

* Update manifest validator

* Update circle

* Update text

* Typo

* fix link to codeowners

* Merge CODEOWNERS into hassfest

* Annotate errors with fixable

* Convert error to warning

* Lint

* Make abs path

* Python 3.5...

* Typo

* Fix tests

bachya committed:
Fix test (#23081)

andrewsayre committed:
HEOS confirm discovered devices before adding (#23063)

* Add host selection step to discovery

* Review feedback

* Fix failing test

rytilahti committed:
Add missing async for tplink's async_setup_platform methods (#23066)

* add missing async for tplink's async_setup_platform methods

thanks to @MartinHjelmare for spotting this, related to #21916

* fix line lengths

fucm committed:
Add support for Stiebel Eltron heat pumps (#21199)

* Start with Stiebel Eltron heatpump

* STE HP

* Add read of operating mode

* Add read-write operation mode

* Further extract ModBus access

* Separation of platform and API

* Last changes

* Use modbus hub

* Update module doc with config

* Clean up platform code

* Cleanup and update to dev2 of pystiebeleltron

* Remove slave configuration

* Add translation of states

* Make name parameter optional

* Consolidate platform

* Correct .coveragerc after conflict

* Prepare component for sensor platform

* Fix issues found in review

* Remove custom states and map to existing HA states

* Force update, when values are modified

* Update CODEOWNERS and requirements_all.txt

* Fix .coveragerc file

* Exclude stiebel_eltron components in .coveragerc

* Break out to module level constant

* Rename constant

* Removed REQ and DEP constant.

bachya committed:
Ensure OpenUV service checks permissions (#22668)

* Create decorator to check service permissions

* Ensure OpenUV service has proper user permissions

* Reverting strange changes

emontnemery committed:
Deprecate implicit state_topic for MQTT discovery (#22998)

* Deprecate implicit state_topic for MQTT discovery

* Lint

* Add comments

* Modernize tests

emontnemery committed:
Fix bugs in MQTT vacuum (#23048)

emontnemery committed:
Refactor MQTT climate to deduplicate code (#23044)

choss committed:
Ignore secrets.yaml when using include_dir_named (#22929)

* ignore secrets.yaml in include_dir_named include

* updating test for include_dir_named secrets ignore

awarecan committed:
Remove validate from aws_config (#23084)

balloob committed:
Migrate packages and check config (#23082)

* Migrate packages and check config

* Fix typing

* Fix check config script

awarecan committed:
Increase timeout (#23098)

RvanBaarle committed:
Add URL query parameters to webhook trigger result data (#23043)

* Added query parameters to webhook data

* Added test for query webhook

* Add second blank line in new test for webhook trigger

pszafer committed:
Add support for params in send_command (#23071)

* add support for params in send_command

* add more tests

balloob committed:
Fix demo (#23087)

* Fix demo

* Fix types

* Fix all the things

* Fix type

* Fix test

* Lint

Adminiuga committed:
Await merge_packages_config(). (#23109)

balloob committed:
Further integration load cleanups (#23104)

* Further integration load cleanups

* Fix tests

* Unflake MQTT vacuum command test

giefca committed:
Fix for stateless covers (#22962)

* Corrections and test

* Raise error if does not support setting position

* Raising error for stateless and assumed state covers

* Fix pylint

* Tests

* Update test_trait.py

cereal2nd committed:
Upgraded python-velbus package, this fixes a lot of problems with the… (#23100)

* Upgraded python-velbus package, this fixes a lot of problems with the velbus component

* updated manifest.json

Kane610 committed:
Support updating deCONZ host address (#22784)

* Update config flow to support updating host address
Improve tests

* Update gateway to handle new address signal

* Improve description why whe need to keep step_init

balloob committed:
Remove loader.get_component (#23111)

* Remove get_component

* Lint

awarecan committed:
Load component which contains data entry flow handler (#23107)

* Load component which contains data entry flow handler

* Use integration.get_component()

* Add error log

Jc2k committed:
Fix homekit_controller climate supported operation_list being blank (#23095)

* Fix tado supported operation modes when used with homekit_controller

* Replace with list comp as requested in review

* More list comps

Kane610 committed:
Axis - start stream when system is ready (#23119)

* Make sure that event stream doesn't start until event listeners are ready

* Change order

JeffLIrion committed:
Handle missing 'serialno' of Android TV (#22996)

* Handle missing 'serialno' for unique ID

* Use None for unique_id if serialno is missing

* Remove name from unique ID

* Use serialno as unique_id

tsvi committed:
Lower verbosity of command line sensor (#23120)

The command line sensor prints every minute the command that is run. This fills up the log.
The command run should be a debug statement.

cgtobi committed:
Bump pyatmo version (#23116)

cgtobi committed:
Clean coveragerc of removed integrations (#23118)

giefca committed:
Add Google Assistant garage type (#23115)

* Tests

* Add Google Assistant GARAGE type

* Update test_trait.py

* Key device class by domain

* Update smart_home.py

pvizeli committed:
Add pvizeli/danielperna84 to code owner of homematic (#22989)

* Add pvizeli/danielperna84 to code owner of homematic

* Update CODEOWNERS

* Run hassfest

mwegrzynek committed:
Add basic Supla cover support (#22133)

* Added basic Supla (https://www.supla.org) support (covers)

* PySupla upgrade, minor spelling corrections and .coveragerc update

* Linter errors cleanup

* More linter cleanups.

* Documentation link removal and import sorting

* Docstring formatting

* PR suggestions

* Styling and linting

* PySupla version update

* Removal of ALLOW_EXTRA in SERVER_CONFIG

* Return False on failed connection validation, function order cleanup

* Component manifest

* Missing return None and different way of setting unique_id

* CODEOWNERS update

* CircleCI nudge

amelchio commented on .coveragerc:
```diff
@@ -164,6 +164,7 @@ omit =
     homeassistant/components/envisalink/*
     homeassistant/components/ephember/climate.py
     homeassistant/components/epson/media_player.py
+    homeassistant/components/epsonworkforceprinter/sensor.py
```
What do you think about naming it "epsonworkforce"? We usually use the product name but not the product category.

amelchio commented on homeassistant/components/epsonworkforceprinter/manifest.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "domain": "epsonworkforceprinter",
+  "name": "Epson Workforce Printer",
+  "documentation": "https://www.home-assistant.io/components/epsonprinter",
```
This naming is inconsistent.

ThaStealth replied:
Yeah, you are correct

amelchio commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+    from epsonprinter_pkg.epsonprinterapi import EpsonPrinterAPI
+    api = EpsonPrinterAPI(host)
+    if not api.available:
+        raise PlatformNotReady("Printer is not available")
```
There is no need for the explanation string.

ThaStealth replied:
Okay, assumed it was required, removed it

amelchio commented on homeassistant/components/epsonworkforceprinter/sensor.py:
```diff
@@ -0,0 +1,85 @@
+class EpsonPrinterCartridge(Entity):
+    """Representation of a cartridge sensor."""
+
+    def __init__(self, hass, api, cartridgeidx):
```
You are not using `hass` (any more).

ThaStealth replied:
Agreed

amelchio reviewed (changes requested):
This looks good, I have just a few nitpicks.

loe committed:
Ensure Boolean configuration values are handled correctly. (#22810)

Kane610 committed:
Axis component support unloading entries (#22692)

* Add support for unloading entries

* Improve config entry tests

* Improve coverage for device

* Remove callback when relevant

pvizeli committed:
Fix ingress bug with Firefox (#23121)

* Fix ingress bug with Firefox

* Fix mock

* Fix tests

* Fix test lint

zewelor committed:
Update yeelight lib (#23123)

* Update yeelight lib

* Run gen requirements

tlyakhov committed:
Update Leviton Decora WiFi library version. (#23125)

fredrike committed:
skip non existing zones (#23113)

balloob committed:
load cleanups (#23112)

* load cleanups

* Remove unused methods

* Allow importing requirements at the top of a file

* Fix test

* Lint

* Install reqs ASAP when loading platforms

Commit:
Fix flux_led only-white controllers (and remove explicit declaration as RGBW in automatic add) (#22210)

* Remove explicit declaration of automatic found devices as RGBW

* fixes for Magic Home only-white controllers

* mode is now set to None instead of removed entirely

* flux_led now changes no values when turned on from off state.

* better checking for changed values in turn_on

* Reduce waiting time to 1 second

* Correction of turn on logic

* Remove accidentally inserted 'not'

* Remove lint

* Remove redundant code

rytilahti committed:
Bump pyhs100 requirement for tplink integration (#23065)

* bump pyhs100 requirement for tplink integration

* Fixes #20994

* Run script/gen_requirements_all.py

balloob committed:
Load integrations only once (#23132)

hunterjm committed:
only preload when stream is setup (#23134)

Kane610 committed:
Axis - improved internal parameter handling (#23122)

Will result in faster startup per entry due to less network data

ThaStealth committed:
renamed Epson Workforce Printer to Epson Workforce

ThaStealth committed:
renamed Epson Workforce Printer to Epson Workforce

ThaStealth committed:
Rebased

ThaStealth committed:
Forgot to update requirements

ThaStealth committed:
Fixed stupid mistake

ThaStealth commented:
So... I messed this PR up, going to create a new one
